build/i2c/src/lib.rs

```diff
@@ -132,8 +169,36 @@ pub enum Disposition {
     /// controller is a target
     Target,
 
-    /// only devices are used (i.e., controller is not used)
+    /// devices are used (i.e., controller is not used), but not as sensors
     Devices,
+
+    /// devices are used, with some used as sensors
+    Sensors,
+}
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
+enum Sensor {
+    Temperature,
+    Power,
+    Current,
+    Voltage,
+    Speed,
+}
+
+impl std::fmt::Display for Sensor {
```

Comment:
Two alternatives to this as an FYI:

- I'd probably implement this as `write!(f, "{:?}", self)` - if the all-caps screamy printing is important, change the enum variant names
- If you want the output in a different case from the variants, consider calling `write_str` in each match arm. I'd put good money on it generating cheaper code. e.g. `Sensor::Temperature => f.write_str("TEMPERATURE"),`